src/test/TestHelper.cpp

```diff
@@ -46,6 +47,26 @@ bool adaptive::AdaptiveTree::download(const char* url,
   return nbRead == 0;
 }
 
+bool TestAdaptiveStream::download(const char* url,
+                                  const std::map<std::string, std::string>& mediaHeaders)
+{
+  size_t nbRead = ~0UL;
+  std::stringstream ss("Sixteen bytes!!!");
+
+  char* buf = (char*)malloc(16);
```

Comment:
Why use c memory management here? And if we have casts they should be c++ casts.

Reply:
I copy/pasted from the implementation in main.cpp, my bad. I'll replace with new/delete